core/convert.go

```diff
@@ -202,6 +202,8 @@ func toRuntimeAPIContainer(c *dockertypes.Container) (*runtimeapi.Container, err
 	if err != nil {
 		return nil, err
 	}
+	// TODO: Currently docker ContainerList api doesn't return annotations, so annotations
+	//       still need to be extracted from labels here.
```

Comment:
I'm not super comfortable with having different sources of truth for annotations depending on the CRI API endpoint queried. It would be straightforward to extend the ContainerList API to return container annotations; perhaps we should do that first so support can be added properly to cri-dockerd?

Reply:
Actually I'm working on it. May I mark this PR a WIP util docker-side supports it ?

Reply:
https://github.com/moby/moby/pull/47866 has been merged into master branch, but released in Pre-release https://github.com/moby/moby/releases/tag/v27.0.0-rc.1. May I bump github.com/docker/docker from 26.1.2+incompatible to v27.0.0-rc.1+incompatible? @corhere